pkg/server/tls_certstore_windows.go

```diff
@@ -102,25 +111,55 @@ func (l *tlsListener) applyWindowsCertificateStore(c TLSConfig) error {
 	return nil
 }
 
-// this is the ASN1 Object Identifier for TemplateID
-var asnTemplateOID = "1.3.6.1.4.1.311.21.7"
+// VerifyPeer is called by the TLS pipeline, and specified in tls.config, this is where any custom verification happens
+func (c *winCertStoreHandler) VerifyPeer(_ [][]byte, verifiedChains [][]*x509.Certificate) error {
+	opts := x509.VerifyOptions{}
+	clientCert := verifiedChains[0][0]
 
-type winCertStoreHandler struct {
-	cfg          WindowsCertificateFilter
-	subjectRegEx *regexp.Regexp
-	log          log.Logger
+	// Check for issuer
```

Comment:
This is the main body that changes the behavior, right?

Reply:
Yes, the checking is now explicitly done in VerifyPeer